app/Models/Concerns/HasFilters.php

```diff
@@ -122,6 +123,8 @@ public function scopeFilter(Builder $query, array $params = []): Builder
                     $this->filterRaces($query, $value);
                 } elseif ($key == 'location_id') {
                     $this->filterLocations($query, $value, $key);
+                } elseif ($key == 'locations_id') {
```

Comment:
Confusing that the variable is singular and the function is plural, and vice versa :) would be cleaner to have a check `if ($this instanceof Race) { filterRaceLocations() }`